packages/flutter/test/rendering/mock_canvas.dart

```diff
@@ -552,6 +557,34 @@ abstract class _TestRecordingCanvasMatcher extends Matcher {
   }
 }
 
+class _TestRecordingCanvasPaintsCountMatcher extends _TestRecordingCanvasMatcher {
+  final Symbol _methodName;
+  final int _count;
+
+  _TestRecordingCanvasPaintsCountMatcher(Symbol methodName, int count)
+      : _methodName = methodName, _count = count;
+
+  @override
+  Description describe(Description description) {
+    return description.add('Object or closure painting $_methodName exactly $_count times');
+  }
+
+  @override
+  bool _evaluatePredicates(Iterable<RecordedInvocation> calls,
+      StringBuffer description) {
+    int count = 0;
+    for(RecordedInvocation call in calls) {
+      if (call.invocation.isMethod && call.invocation.memberName == _methodName) {
+        count++;
+      }
+    }
+    if (count != _count) {
+      throw 'It painted $_methodName $count times instead of $_count times.';
```

Comment:
I believe _evaluatePredicates should just return false on a miss-match and you add the failure reason to the StringBuffer `description`.

Reply:
Done.